homeassistant/components/stt/legacy.py

```diff
@@ -27,6 +27,18 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+@callback
+def async_default_provider(hass: HomeAssistant) -> str | None:
+    """Return the domain of the default provider."""
+    if not hass.data[DATA_PROVIDERS]:
+        return None
+
+    if "cloud" in hass.data[DATA_PROVIDERS]:
+        return "cloud"
+
+    return next(iter(hass.data[DATA_PROVIDERS]))
```

Comment:
Alternatively we can drop the first check at the top of the body and add a default here.
```suggestion
    return next(iter(hass.data[DATA_PROVIDERS]), None)
```